test/functional/feature_utxo_set_hash.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test UTXO set hash value calculation.
+
+Based on the UTXO set of a P2P connection serialize and calculate a hash
+for the UTXO set and compare it with the actual node's hash.
+"""
+
+from test_framework.blocktools import (
+    create_block,
+    create_coinbase,
+    create_tx_with_script,
+)
+from test_framework.messages import (
+    COIN,
+)
+from test_framework.p2p import P2PDataStore
+from test_framework.muhash import MuHash3072
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+import hashlib
+import struct
+
+class UTXOSetHashTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def run_test(self):
+        node = self.nodes[0]
+        conn = node.add_p2p_connection(P2PDataStore())
+
+        # Get tip information (Genesis) to build the next block.
+        best_block_hash = node.getbestblockhash()
+        tip = int(best_block_hash, 16)
+        best_block = node.getblock(best_block_hash)
+        height = best_block["height"] + 1
+        block_time = best_block["time"] + 1
+
+        # Generate 100 blocks so the first coinbase is matured. We have to do
+        # it in a way that conn knows about them so we use
+        # send_blocks_and_test.
+        for i in range(100):
+            # Coinbases are anyone can spend.
+            block = create_block(tip, create_coinbase(height), block_time)
+            block.solve()
+            conn.send_blocks_and_test([block], node, success=True)
+
+            tip = block.sha256
+            best_block = block
+            height += 1
+            block_time += 1
+
+            # Save first block because we want to spend its coinbase.
+            if i == 0:
+                block1 = block
```

Comment:
It was and I did it because I thought there was a good chance that this code could be used for other tests as well and it also made some things a bit easier. But overall of course it adds complexity and I am lacking a specific example where it would be used so I have removed the P2P based implementation and simplified it in a similar way as you imagined, I think. Please check it out.